pkg/api/admin/openshiftcluster.go

```diff
@@ -149,8 +149,9 @@ type ConsoleProfile struct {
 
 // ServicePrincipalProfile represents a service principal profile.
 type ServicePrincipalProfile struct {
-	ClientID   string `json:"clientId,omitempty"`
-	SPObjectID string `json:"spObjectId,omitempty"`
+	ClientID     string `json:"clientId,omitempty"`
+	SPObjectID   string `json:"spObjectId,omitempty"`
+	ClientSecret string `json:"clientSecret,omitempty"`
```

Comment:
Since service principal is a pointer now (and we have to recreate the object if not nil rather than write directly to sub-fields), during testing it looked like we were overwriting clientsecret during an admin api action such as PUCM because we were re-populating ServicePrincipalProfile with only ClientID and SPObjectID. In the past, we didn't need clientsecret in the adminAPI because we were writing directly to the existing fields rather than creating a new api.ServicePrincipalProfile object and passing that back to cosmosdb.